### PR TITLE
Dh admin panel

### PR DESCRIPTION
**Pull request by Dimo-Hristov**
1. Add interfaces
2. Improve architecture
3. Improve routing
4. Create services and signals. and data sharing.
5. Create shared components. Cards list for events and table for users.
6. Create missing pages and check for all redirects and parent-child rendering.
7. Make the main views responsive, add hamburger menu and event-card list .  (waiting for the design to finish the hamburger menu)
8. Add todos for the pages i have started and havent finish.

### CODE REVIEW

File: src/app/components/core/hero/hero.component.html [hunk 1]
```diff
@@ -1,4 +1,6 @@
 <section class="hero">
   <h1>Административен панел</h1>
-  <img src="../../../assets/icons/hero-wheels.png" alt="asdasd" />
+  <div class="image-container">
+    <img src="../../../assets/icons/hero-wheels.png" alt="asdasd" />
```
Comment (Getsov) [inline src/app/components/core/hero/hero.component.html:4]: alt="asdasd" ? And why there is an image in "icons" folder?

File: src/enviroments/enviroment.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const environment = {
```
Comment (Getsov) [inline src/enviroments/enviroment.ts:1]: this file must be in git ignore

File: src/shared/components/accounts-table/accounts-table.component.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
+<table class="accounts-table">
+  <thead>
+    <tr>
+      <th>
+        <input type="checkbox" />
+      </th>
+      <th>
+        Имейл
+        <img src="../../../../assets/icons/table-arrows.png" alt="sort icon" />
```
Comment (Getsov) [inline src/shared/components/accounts-table/accounts-table.component.html:9]: why such "icon" with image? There are icons for this.

File: src/shared/components/accounts-table/accounts-table.component.html [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      </th>
+      <th>
+        Име<img
+          src="../../../../assets/icons/table-arrows.png"
```
Comment (Getsov) [inline src/shared/components/accounts-table/accounts-table.component.html:13]: Same here for the "icon-image"

File: src/shared/components/accounts-table/accounts-table.component.html [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+          alt="sort icon"
+        />
+      </th>
+      <th>
+        Телефон<img
+          src="../../../../assets/icons/table-arrows.png"
```
Comment (Getsov) [inline src/shared/components/accounts-table/accounts-table.component.html:19]: Same here for the "icon-image"

File: src/shared/components/accounts-table/accounts-table.component.html [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+          alt="sort icon"
+        />
+      </th>
+      <th>
+        Регион<img
+          src="../../../../assets/icons/table-arrows.png"
```
Comment (Getsov) [inline src/shared/components/accounts-table/accounts-table.component.html:25]: Same here for the "icon-image"

File: src/shared/components/accounts-table/accounts-table.component.html [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    @for (user of usersList; track user._id) {
+    <tr>
+      <td><input type="checkbox" /></td>
+      <td>{{ user.email }}</td>
+      <td>{{ user.firstName }} {{ user.lastName }}</td>
+      <td>{{ user.phone }}</td>
+      <td>{{ user.region }}</td>
+      <td class="buttons">
+        <a routerLink="/user/{{ user._id }}/details">
+          <img src="../../../../assets/icons/table-view.png" alt="View" />
```
Comment (Getsov) [inline src/shared/components/accounts-table/accounts-table.component.html:42]: This should be a button, not an image

File: src/shared/components/accounts-table/accounts-table.component.html [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        </a>
+        <a routerLink="/user/{{ user._id }}/edit">
+          <img src="../../../../assets/icons/table-edit.png" alt="Edit" />
```
Comment (Getsov) [inline src/shared/components/accounts-table/accounts-table.component.html:45]: This should be a button, not an image

File: src/shared/components/accounts-table/accounts-table.component.html [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        </a>
+        <a>
+          <img src="../../../../assets/icons/table-approve.png" alt="Approve" />
```
Comment (Getsov) [inline src/shared/components/accounts-table/accounts-table.component.html:48]: This should be a button, not an image

File: src/shared/components/accounts-table/accounts-table.component.html [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+        </a>
+        <a>
+          <img src="../../../../assets/icons/table-delete.png" alt="Delete" />
```
Comment (Getsov) [inline src/shared/components/accounts-table/accounts-table.component.html:51]: This should be a button, not an image

File: src/shared/services/user.service.ts [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+  // SIGNALS --------
+
+  // Organizers for approval
+
+  organizersForApproval = signal<User[]>([]);
+
+  hasOrganizersForapproval: boolean = this.organizersForApproval().length < 1;
+
+  setOrganizersForApprove() {
```
Comment (Getsov) [inline src/shared/services/user.service.ts:41]: get, not set

File: src/app/app.routes.ts [hunk 12]
```diff
@@ -1,6 +1,6 @@
 import { Routes } from '@angular/router';
 import { EventsComponent } from './components/events/events.component';
-import { CreateEventComponent } from './components/events/create-event/create-event.component';
+import { CreateEventComponent } from './components/event/create-event/create-event.component';
```
Comment (Getsov) [inline src/app/app.routes.ts:3]: the main folder should be "events", as it was, not "event"

File: src/shared/interfaces/Event.ts [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import { User } from './User';
+
+export interface Event {
```
Comment (Getsov) [inline src/shared/interfaces/Event.ts:3]: Double check the interfaces with the app and the back end. There are mistakes, like "likes" is optional and etc. Check them all

File: src/shared/services/user.service.ts (hunk 11 above)
Comment (Dimo-Hristov) [reply src/shared/services/user.service.ts:41]: I have two different functions, the first in the api call is getOrganizers the second function setOrganizers is then we need to set them in the state.

File: src/shared/components/accounts-table/accounts-table.component.html (hunk 10 above)
Comment (Dimo-Hristov) [reply src/shared/components/accounts-table/accounts-table.component.html]: done

File: src/shared/components/accounts-table/accounts-table.component.html (hunk 9 above)
Comment (Dimo-Hristov) [reply src/shared/components/accounts-table/accounts-table.component.html]: done

File: src/shared/components/accounts-table/accounts-table.component.html (hunk 5 above)
Comment (Dimo-Hristov) [reply src/shared/components/accounts-table/accounts-table.component.html]: done

File: src/shared/components/accounts-table/accounts-table.component.html (hunk 4 above)
Comment (Dimo-Hristov) [reply src/shared/components/accounts-table/accounts-table.component.html]: done

File: src/shared/components/accounts-table/accounts-table.component.html (hunk 3 above)
Comment (Dimo-Hristov) [reply src/shared/components/accounts-table/accounts-table.component.html]: done

File: src/enviroments/enviroment.ts (hunk 2 above)
Comment (Dimo-Hristov) [reply src/enviroments/enviroment.ts:1]: done

File: src/app/components/core/hero/hero.component.html (hunk 1 above)
Comment (Dimo-Hristov) [reply src/app/components/core/hero/hero.component.html]: fixed

File: src/app/app.routes.ts (hunk 12 above)
Comment (Dimo-Hristov) [reply src/app/app.routes.ts:3]: As we talk at the meeting, we will not change it .

**Review by Getsov — APPROVED** [review]

**Pull request merged**